### PR TITLE
ci: DRA enhancements

### PR DESCRIPTION
**Pull request by v1v**
### What

- Add who to contact with if a DRA failure or a package failure.
- Use releaseManager step
- Add DRA build failure analyser.

### Issue

Requires https://github.com/elastic/apm-pipeline-library/pull/1628

### Test

See [this build](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackaging/detail/PR-31073/7/pipeline/)

image

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -22,9 +22,9 @@ pipeline {
     DOCKER_REGISTRY = 'docker.elastic.co'
     GITHUB_CHECK_E2E_TESTS_NAME = 'E2E Tests'
     SNAPSHOT = "true"
-    PIPELINE_LOG_LEVEL = "INFO"
-    SLACK_CHANNEL = '#beats'
-    NOTIFY_TO = 'beats-contrib+package-beats@elastic.co'
+    PIPELINE_LOG_LEVEL = "DEBUG"
+    SLACK_CHANNEL = 'UJ2J1AZV2'
+    NOTIFY_TO = 'victor.martinez+package-beats@elastic.co'
```
Comment (v1v) [inline .ci/packaging.groovy:27]:
```suggestion
    PIPELINE_LOG_LEVEL = "INFO"
    SLACK_CHANNEL = '#beats'
    NOTIFY_TO = 'beats-contrib+package-beats@elastic.co'
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -42,12 +42,15 @@ pipeline {
...
     stage('Filter build') {
+      options { skipDefaultCheckout() }
       agent { label 'ubuntu-20 && immutable' }
       when {
         beforeAgent true
         anyOf {
           triggeredBy cause: "IssueCommentCause"
           expression {
+            // TODO: for testing purposes
+            return true
```
Comment (v1v) [inline .ci/packaging.groovy:53]:
```suggestion
```

File: .ci/packaging.groovy [hunk 3]
```diff
@@ -90,7 +93,9 @@ pipeline {
             dir("${BASE_DIR}"){
               setEnvVar('BEAT_VERSION', sh(label: 'Get beat version', script: 'make get-version', returnStdout: true)?.trim())
             }
-            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
+            //TODO: for testing purposes
+            //setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
+            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable('main'))
```
Comment (v1v) [inline .ci/packaging.groovy:98]:
```suggestion
            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
```

File: .ci/packaging.groovy [hunk 4]
```diff
@@ -100,12 +105,15 @@ pipeline {
           }
         }
         stage('Run E2E Tests for Packages'){
+          // TODO: for testing purposes
+          when { expression { return false } }
```
Comment (v1v) [inline .ci/packaging.groovy:109]:
```suggestion
```

File: .ci/packaging.groovy [hunk 5]
```diff
@@ -124,17 +133,24 @@ pipeline {
...
               }
               sh(label: "debug package", script: 'find build/distributions -type f -ls || true')
+              sh(label: 'prepare-release-manager-artifacts', script: ".ci/scripts/prepare-release-manager.sh")
               dockerLogin(secret: env.DOCKERELASTIC_SECRET, registry: env.DOCKER_REGISTRY)
-              script {
-                getVaultSecret.readSecretWrapper {
-                  sh(label: 'release-manager.sh', script: '.ci/scripts/release-manager.sh')
-                }
+              //TODO: for testing purposes
+              withEnv(["BRANCH_NAME=main"]){
```
Comment (v1v) [inline .ci/packaging.groovy:139]:
```suggestion
```

File: .ci/packaging.groovy [hunk 6]
```diff
@@ -124,17 +133,24 @@ pipeline {
...
+              releaseManager(project: 'beats',
+                             version: env.BEAT_VERSION,
+                             type: 'snapshot',
+                             artifactsFolder: 'build/distributions',
+                             outputFile: env.DRA_OUTPUT)
               }
```
Comment (v1v) [inline .ci/packaging.groovy:145]:
```suggestion
                             outputFile: env.DRA_OUTPUT)
```

File: .ci/packaging.groovy [hunk 7]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@test/releaseManager') _
```
Comment (v1v) [inline .ci/packaging.groovy:3]:
```suggestion
@Library('apm@current') _
```

File: .ci/packaging.groovy [hunk 8]
```diff
@@ -124,17 +127,21 @@ pipeline {
...
+                             outputFile: env.DRA_OUTPUT)
             }
           }
           post {
             failure {
-              notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}@${env.BRANCH_NAME}] DRA failed", body: "Build: (<${env.RUN_DISPLAY_URL}|here>)")
+              notifyStatus(analyse: true,
+                           file: "${BASE_DIR}/${env.DRA_OUTPUT}",
+                           subject: "[${env.REPO}@${env.BRANCH_NAME}] The Daily releasable artifact failed.",
+                           body: 'Contact the Release Platform team [#platform-release]')
```
Comment (ph) [inline .ci/packaging.groovy:144]: @v1v I presume our build is not great enough top has summary of why it failed?

**Review by kuisathaverat — APPROVED** [review]

**Review by ph — APPROVED** [review]

**Pull request merged**